src/base/record.rs

```diff
@@ -182,6 +183,27 @@ impl<N, D> From<(N, u32, D)> for Record<N, D> {
 }
 
 
+//--- OctetsFrom
+//
+// XXX We don’t have blanket FromOctets for a type T into itself, so this may
+//     not always work as expected. Not sure what we can do about it?
+
+impl<Name, Data, SrcName, SrcData>
+OctetsFrom<Record<SrcName, SrcData>> for Record<Name, Data>
+where Name: OctetsFrom<SrcName>, Data: OctetsFrom<SrcData> {
+    fn octets_from(
+        source: Record<SrcName, SrcData>
+    ) -> Result<Self, ShortBuf> {
+        Ok(Record {
+            owner: Name::octets_from(source.owner)?,
+            class: source.class,
+            ttl: source.ttl,
+            data: Data::octets_from(source.data)?
+        })
+    }
+}
```

Comment:
Maybe add an implementation for `N: SrcName` instead? SrcName should have `ToDname`, so converting an `ParsedRecord<impl OctetsRef>` into `Record<Dname<O>, ...>` should be possible.

Reply:
I feel that this sort of thing should happen via regular `From` from a `ParsedDname` to its `Dname` given that it actually copies things around. The intention of `OctetsFrom` was really just a type conversion for the particular octet type.

Reply:
Yes that's fair, however scanning ParsedRecord and converting is common and currently very clunky.

```rust
let rr = rr.into_record::<Dnskey<_>>()?.context("empty record")?;
let rr: Record<Dname<O>, Dnskey<O>> = Record::new(rr.owner().to_dname().context("failed to convert dname")?, rr.class(), rr.ttl(), rr.into_data().octets_into().context("failed to convert data")?);
```

If we add `impl From<ParsedDname<R>> for Dname<O>` that should make `Record::from_record` work, another option is to move this into another method/trait.

Reply:
I think there indeed lots of `From` impls and other conversion helpers missing. My plan was to add them as they become necessary so as to not add too much stuff that nobody uses and that just increases the code size.

I suggest we look at conversions for records and names in a separate PR.

Reply:
Yes, of course, I didn't mean to hold this up, sorry.

Reply:
Nono! I really appreciate the feedback. Here is as good as anywhere.